src/main/java/org/wso2/carbon/esb/connector/MQConnectionBuilder.java

```diff
@@ -0,0 +1,218 @@
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+package org.wso2.carbon.esb.connector;
+
+import com.ibm.mq.MQEnvironment;
+import com.ibm.mq.MQException;
+import com.ibm.mq.MQQueueManager;
+import com.ibm.mq.MQSimpleConnectionManager;
+import com.ibm.mq.constants.CMQC;
+import com.ibm.mq.constants.CMQXC;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.MessageContext;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManagerFactory;
+import java.io.FileInputStream;
+import java.security.KeyStore;
+import java.util.*;
+import java.util.concurrent.*;
+
+/**
+ * Start connection with IBM MQ queue manager
+ */
+public class MQConnectionBuilder {
+
+    private static final Log logger = LogFactory.getLog(MQConnectionBuilder.class);
+
+    private MQQueueManager queueManager;
+
+    private MQConfiguration config;
+
+    MQConnectionBuilder(MessageContext msg) {
+
+        this.config = new MQConfiguration(msg);
+        if (config.getCiphersuit().contains("TLS")) {
+            Properties props = System.getProperties();
+            props.setProperty("com.ibm.mq.cfg.useIBMCipherMappings", "false");
+        }
+        if (config.isSslEnable()) {
+            MQEnvironment.sslCipherSuite = config.getCiphersuit();
+            MQEnvironment.sslSocketFactory = createSSLContext().getSocketFactory();
+            MQEnvironment.sslCipherSuite = config.getCiphersuit();
+            MQEnvironment.sslFipsRequired = config.getFlipRequired();
+        }
+        MQEnvironment.properties.put(CMQC.TRANSPORT_PROPERTY, CMQC.TRANSPORT_MQSERIES_CLIENT);
+        MQEnvironment.properties.put(CMQC.USER_ID_PROPERTY, config.getUserName());
+        MQEnvironment.properties.put(CMQC.PASSWORD_PROPERTY, config.getPassword());
+        Collection headerComp = new Vector();
+        headerComp.add(new Integer(CMQXC.MQCOMPRESS_SYSTEM));
+        MQEnvironment.hdrCompList = headerComp;
+        MQEnvironment.setDefaultConnectionManager(customizedPool(config.getTimeout(), config.getmaxConnections(), config.getmaxnusedConnections()));
+
+        List<String> hostandportList;
+        List<String> channelList;
+
+        if (config.getReconnectList() != null) {
+            hostandportList = config.getReconnectList();
+        } else {
+            hostandportList = new ArrayList<>();
+        }
+
+        if (config.getChannelList() != null) {
+            channelList = config.getChannelList();
+        } else {
+            channelList = new ArrayList<>();
+        }
+
+        hostandportList.add(0, config.getHost() + "/" + config.getPort());
+        channelList.add(0, config.getChannel());
+
+        if (queueManager == null || !queueManager.isConnected()) {
+            queueManager = getQueueManager(config.getHost(), config.getChannel(), config.getPort());
```

Comment:
Please improve logging for,
[1] retry with retry interval.
[2] Successful queue manager return.
[3] Un-Successful queue manager return.